shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/domain/usecase/GetUserProfileUseCase.kt

```diff
@@ -0,0 +1,19 @@
+package co.nimblehq.avishek.phong.kmmic.domain.usecase
+
+import co.nimblehq.avishek.phong.kmmic.domain.model.User
+import co.nimblehq.avishek.phong.kmmic.domain.repository.UserRepository
+import kotlinx.coroutines.flow.Flow
+
+interface GetUserProfileUseCase {
+    operator fun invoke(): Flow<User>
+}
+
+class GetUserProfileUseCaseImpl(
+    private val userRepository: UserRepository
+): GetUserProfileUseCase {
+
+    override operator fun invoke(): Flow<User> {
+        return userRepository
+            .getProfile()
```

Comment:
I think we can write it in one line. We can do that in places where there's only one dot call in the chain.
```suggestion
        return userRepository.getProfile()
```